index.md

```diff
@@ -1,213 +1,297 @@
+        <h1>Modern, reactive framework</h1>
+
+        <p>
+        Easily compose your UI with Flutter's
+        modern functional-reactive framework and
+        rich set of platform, layout, and foundation widgets.
+        Solve your tough UI challenges with
+        powerful and flexible APIs for 2D, animation, gestures,
+        effects, and more.
+        </p>
+
+{% prettify dart %}
+class CounterState extends State<Counter> {
+  int counter = 0;
+
+  void increment() {
+    // Tells the Flutter framework that state has changed,
+    // so the framework can run build() and update the display.
+    setState(() {
+      counter++;
+    });
+  }
+
+  Widget build(BuildContext context) {
+    // This method is rerun every time setState is called.
+    // The Flutter framework has been optimized to make rerunning
+    // build methods fast, so that you can just rebuild anything that
+    // needs updating rather than having to individually change
+    // instances of widgets.
+    return new Row(
+      children: <Widget>[
+        new RaisedButton(
+          onPressed: increment,
+          child: new Text('Increment'),
+        ),
+        new Text('Count: $counter'),
+      ],
+    );
+  }
+}
+{% endprettify %}
+
+        <p>
+        Browse the <a href="/widgets/">widget catalog</a>
+        and learn more about the
+        <a href="/widgets-intro/">functional-reactive framework</a>.
+        </p>
+
+    </section>
+
+    <section class="homepage__interop">
+        <h1>Platform interop and native code</h1>
+
+        <p>
+        Make your app come to life
+        with platform APIs, 3rd party SDKs,
+        and native code.
+        Reuse your existing Java, Swift, and ObjC code.
+        </p>
+
+{% prettify dart %}
+// From https://github.com/flutter/flutter/tree/master/examples/platform_channel
+
+Future<Null> getBatteryLevel() async {
+  var batteryLevel = 'unknown';
+  try {
+    int result = await methodChannel.invokeMethod('getBatteryLevel');
+    batteryLevel = 'Battery level: $result%';
+  } on PlatformException {
+    batteryLevel = 'Failed to get battery level.';
+  }
+  setState(() {
+    _batteryLevel = batteryLevel;
+  });
+}
+{% endprettify %}
+
+        <p>
+        Learn how to write a <a href="/platform-plugins/">platform plugin</a>
+        to access native code and APIs.
+        </p>
+
+    </section>
+
+    <section class="homepage__features">
+        <h1>Unified app development</h1>
+
+        <p>
+        Flutter has the tools and libraries to help you easily
+        bring your ideas to life on iOS and Android.
+        If you don't have any mobile development experience, Flutter
+        is an easy and fast way to build beautiful mobile apps.
+        If you are an experienced iOS or Android developer,
+        you can use Flutter for your views and leverage much of your
+        existing Java/ObjC/Swift investment.
+        </p>
+
+        <div class="feature-lists">
```

Comment:
One insight from this exercise is that we have a ton of awesome capabilities in our system, but we don't always have docs that link to the specific feature or capability. Agreed that many of these items can and should be links. I'll try to find links for the subjects that do have a natural documentation for them.